how-to/customize-workspace/client/src/modules/analytics/example/analytics-module.ts

```diff
@@ -0,0 +1,79 @@
+import type { Logger, LoggerCreator } from "customize-workspace/shapes";
+import type { AnalyticsModule, PlatformAnalyticsEvent } from "customize-workspace/shapes/analytics-shapes";
+import type { ModuleDefinition, ModuleHelpers } from "customize-workspace/shapes/module-shapes";
+import type { ExampleConsoleAnalyticsOptions } from "./shapes";
+
+/**
+ * Implement the log provider using the console.
+ */
+export class ExampleConsoleAnalyticsModule implements AnalyticsModule<ExampleConsoleAnalyticsOptions> {
+	private _logger: Logger;
+
+	private _logEvent: (message: string, events: PlatformAnalyticsEvent[]) => void;
+
+	/**
+	 * Initialize the module.
+	 * @param definition The definition of the module from configuration include custom options.
+	 * @param loggerCreator For logging entries.
+	 * @param helpers Helper methods for the module to interact with the application core.
+	 * @returns Nothing.
+	 */
+	public async initialize(
+		definition: ModuleDefinition<ExampleConsoleAnalyticsOptions>,
+		loggerCreator: LoggerCreator,
+		helpers: ModuleHelpers
+	): Promise<void> {
+		this._logger = loggerCreator("ExampleConsoleAnalyticsModule");
+		this._logger.info("ExampleConsoleAnalyticsModule initialized");
+		this._logger.info("Session Id: ", helpers.sessionId);
+		const logLevel = definition?.data?.eventLogLevel ?? "info";
+		switch (logLevel) {
+			case "trace": {
+				this._logEvent = (message, events) => {
```

Comment:
This could all be simplified to

```
this._logEvent = (message, events) => {
	// we don't want to trace the function so we log it as debug/verbose
	this._logger[logLevel === "trace" ? "debug" : logLevel](message, JSON.stringify(events));
};
```

Reply:
That still means we are doing the check every time an event comes in. We can capture the level out of the function so the check is only done once.